### ISSUE TITLE
Filtering for State Province is confusing

### ISSUE THREAD

**Issue by ManonGros** [gbif/portal-feedback#3316]
One of our users was having trouble using the UI to select records with the State Province "Florida".
And I agree: there are [37,992,233 records with the State Province "Florida"](https://www.gbif.org/occurrence/search?state_province=Florida&advanced=1) but when looking for "Florida" in the UI, all the other suggestions appear first:
<img width="390" alt="Screenshot 2021-03-11 at 09 41 25" src="https://user-images.githubusercontent.com/7677271/110759503-f8037180-824d-11eb-8ae8-8e82f6fc73c0.png">

So there are 3 things that make the search confusing:
1. The suggested values for the State Province aren't ordered by number of records (nor Alphabetic order?) and since there are only 10 suggested value, the main one doesn't show.
2. If I press enter after looking for "Florida", the UI selects the first suggestion ("FLORIDA COLLIER CO.") instead of what I typed so I can't just look for "Florida" (I had to put "Florida" in the URL to get what I needed).
3. Once a value is selected, it remains in the suggestions (see screenshot below) so it doesn't seem like it is possible to select anything else than the first 10 values suggested.

<img width="360" alt="Screenshot 2021-03-11 at 09 45 14" src="https://user-images.githubusercontent.com/7677271/110760076-9bed1d00-824e-11eb-9ed0-6b48ab01d87c.png">

**Comment by MortenHofft** [comment]
The user should probably use the GADM filter instead.
https://www.gbif.org/occurrence/search?advanced=1&gadm_gid=USA.10_1

Suggest order is an API concern, but it makes sense to have a perfect match as the top suggestion.

For now the only way around it in the UI is to click escape and press enter. Then it will apply whatever text you entered. We could of course ask for more suggestions, but that seems more like a bandaid than a solution to me. Perhaps the library can be configured to not preselect the first option, that might help for free text search filters

**Comment by ManonGros** [comment]
Thanks @MortenHofft!
This user wanted to include the records that didn't have any coordinates so that's why he wasn't using the GADM filter.
Could there be a way to remove from the suggestions, the values that have already selected? (like for the scientific name filter)

**Comment by MortenHofft** [comment]
the scientific name filters removes suggestions client side. So the result set becomes smaller and smaller everytime you select something. The results are the same. I'm not personally a fan of that addition.

Try to search for scientific name `puma` on [this page](https://www.gbif.org/occurrence/search?taxon_key=2435098&taxon_key=6164622&taxon_key=2435099&taxon_key=2154611&taxon_key=9177759&taxon_key=1533747&taxon_key=2435146&taxon_key=1137360&taxon_key=7422339&taxon_key=9812093&occurrence_status=present)

You will get no suggestions. Despite there being [many more options](https://api.gbif.org/v1/species/suggest?datasetKey=d7dddbf4-2cf0-4f39-9b2a-bb099caae36c&limit=50&q=puma)

I see the need, but do not believe it is the best solution to the problem. We could increase the suggest limit everytime someone had something in their filter already. But that isn't good either as the existing query could be for something else and we would just ask for more and more. We could have more suggestions, but where to end. I do not see any good client side solutions to this. the API should just return what you search for if it exists. At least I cannot imagine a solution to this that is about UI

**Comment by ManonGros** [comment]
In that case, perhaps we should just remove the suggestions for that filter? It would be less confusing.

**Comment by MortenHofft** [comment]
Yes we can remove it. But I'm not sure I think it is a better solution. The suggestions help you understand the data, without it, it is hard to know what is in there.

**Comment by ManonGros** [comment]
True, then perhaps there should be some API change.
As you mentioned before, it would be good if the API could return a perfect match at the top of the search.

**Comment by MortenHofft** [comment]
And not differentiate between "Copenhagen" and "copenhagen" in suggestions when search ignores the casing anyhow
